### DISCUSSION TITLE
Token Invalid error

### DISCUSSION THREAD

**Discussion by alsongarbuja** [fattureincloud/api#445] [Q&A] [answered]
Hello,

I am currently trying to use fattureincloud to manage invoices for a client project but I am getting the following error, `(invalid token)` when trying to get the `access_token` through OAuth access flow.

```js
Error: {"error":"invalid_grant","error_description":"Invalid token."}
```
I tried searching for what the error could mean but since my first language is not italian nor english it is little hard for me to get resources. 
I am using NEXT JS to create the whole frontend and backend.

```js
try {
      const oauth = getOauthCodeManager();
      const params = oauth.getParamsFromUrl(req.body.url);
      const auth_code = params.authorizationCode;
      const token = await oauth.fetchToken(auth_code); // error is occuring here
      res.json({ token });
} catch (error) {
      console.log(error);
      res.json({ error: error });
}
```

Can anyone explain me why this error is ocurring?

**Comment by valmoz** [comment]
Hi @alsongarbuja 
don't worry, we'll try to support you as much as possible. Unfortunately, we can only provide you support in English.

I think the problem could be the auth_code variable's content. Can you please try to print the variable and check if it contains a string starting with "a/"?

Thanks
Mauro

**Reply by alsongarbuja** [reply]
I have logged the authorization code and the code starts with "c/" which is what your [documentation says](https://developers.fattureincloud.it/docs/authentication/code-flow/sdk-oauth-helper/#key-token-types)

I did found a disuccsion where you said to check if the code starts with "a/" and I tried to change the code to that with string mainpulation before sending it to `await oauth.fetchToken(..)` function. I got error of wrong token type.

**Reply by valmoz** [reply, marked as answer]
Sorry @alsongarbuja my mistake, "c/" is the correct prefix.

If the auth code has the correct value, it should work correctly. From our logs it seems that you're executing the same operation twice, the first one goes to success and returns the token, invalidating the auth code. When you try to perform the same operation again with the same code you obtain 401 error because the code was invalidated at the first attempt.
You should try to check why your code is performing the same operation twice in a short amount of time.

Thanks 
Mauro

**Reply by alsongarbuja** [reply]
Turns out react strict mode was running the `useEffect` twice.

It's working.

Thanks @valmoz

**Reply by valmoz** [reply]
I'm glad that you solved it.
Thanks
Mauro
